src/SDL_hashtable.h

```diff
@@ -0,0 +1,64 @@
+/*
+  Simple DirectMedia Layer
+  Copyright (C) 1997-2024 Sam Lantinga <slouken@libsdl.org>
+
+  This software is provided 'as-is', without any express or implied
+  warranty.  In no event will the authors be held liable for any damages
+  arising from the use of this software.
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely, subject to the following restrictions:
+
+  1. The origin of this software must not be misrepresented; you must not
+     claim that you wrote the original software. If you use this software
+     in a product, an acknowledgment in the product documentation would be
+     appreciated but is not required.
+  2. Altered source versions must be plainly marked as such, and must not be
+     misrepresented as being the original software.
+  3. This notice may not be removed or altered from any source distribution.
+*/
+#ifndef SDL_hashtable_h_
+#define SDL_hashtable_h_
+
+// this is not (currently) a public API. But maybe it should be!
```

Comment:
Do we want to make this public in SDL's API so we don't have to copy it to other projects?

Reply:
I was debating that, but it's SO ugly to use. You have to cast every access and insert. I'm thinking that people who really need it can copy it, but I could be convinced either way.

Reply:
Most projects will have std::map or something similar available to them.

Reply:
Okay, let's leave it alone. I'm glad this got picked up _after_ the new hashtable code arrived, though!

Reply:
Yep, me too. :)